http-netty/src/main/java/io/micronaut/http/netty/websocket/AbstractNettyWebSocketHandler.java

```diff
@@ -315,9 +321,31 @@ protected void handleWebSocketFrame(ChannelHandlerContext ctx, WebSocketFrame ms
                         CloseReason.UNSUPPORTED_DATA
                 );
             } else {
+                ByteBuf msgContent = msg.content().retain();
+                if (!msg.isFinalFragment()) {
+                    frameBuffer.updateAndGet((buffer) -> {
+                        if (buffer == null) {
+                            buffer = UnpooledByteBufAllocator.DEFAULT.compositeBuffer();
```

Comment:
@graemerocher If there is a different allocator I should be using here please lmk

Reply:
Should use `ctx.alloc().compositeBuffer()`